extensions/azure-cosmos/runtime/src/main/java/io/quarkiverse/azure/cosmos/runtime/CosmosClientProducer.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkiverse.azure.cosmos.runtime;
+
+import jakarta.enterprise.inject.Produces;
+import jakarta.inject.Inject;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosClient;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+
+import io.quarkiverse.azure.core.util.AzureQuarkusIdentifier;
+
+public class CosmosClientProducer {
+
+    @Inject
+    CosmosConfig cosmosConfiguration;
+
+    @Produces
+    public CosmosClient createCosmosClient() {
+        CosmosClientBuilder builder = getBuilder();
+        return null == builder ? null : builder.buildClient();
+    }
+
+    @Produces
+    public CosmosAsyncClient createCosmosAsyncClient() {
+        CosmosClientBuilder builder = getBuilder();
+        return null == builder ? null : builder.buildAsyncClient();
+    }
+
+    private CosmosClientBuilder getBuilder() {
+        if (!cosmosConfiguration.enabled) {
+            return null;
+        }
+
+        assert cosmosConfiguration.endpoint.isPresent() : "The endpoint of Azure Cosmos DB must be set";
+        CosmosClientBuilder builder = new CosmosClientBuilder()
```

Comment:
What happens with an invalid endpoint?

Reply:
The app won't be started as the invoking of `createCosmosClient` or `createCosmosAsyncClient` will fail with related error message.